crates/bevy_ecs/src/schedule/stepping.rs

```diff
@@ -0,0 +1,1409 @@
+        stepping
+            .add_schedule(TestSchedule)
+            .enable()
+            .set_breakpoint(TestSchedule, second_system)
+            .continue_frame();
+
+        assert_schedule_runs!(&schedule, &mut stepping, first_system);
+        stepping.continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping, second_system);
+
+        stepping.clear_breakpoint(TestSchedule, second_system);
+        stepping.continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping, first_system, second_system);
+    }
+
+    #[test]
+    fn clear_system() {
+        let (schedule, _world) = setup();
+
+        let mut stepping = Stepping::new();
+        stepping
+            .add_schedule(TestSchedule)
+            .enable()
+            .never_run(TestSchedule, second_system)
+            .continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping, first_system);
+
+        stepping.clear_system(TestSchedule, second_system);
+        stepping.continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping, first_system, second_system);
+    }
+
+    #[test]
+    fn clear_schedule() {
+        let (schedule, _world) = setup();
+
+        let mut stepping = Stepping::new();
+        stepping
+            .add_schedule(TestSchedule)
+            .enable()
+            .never_run(TestSchedule, first_system)
+            .never_run(TestSchedule, second_system)
+            .continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping,);
+
+        stepping.clear_schedule(TestSchedule);
+        stepping.continue_frame();
+        assert_schedule_runs!(&schedule, &mut stepping, first_system, second_system);
+    }
+
+    // Schedules such as FixedUpdate can be called multiple times in a single
+    // render frame.  Ensure we only run steppable systems the first time the
+    // schedule is run
+    #[test]
+    fn multiple_calls_per_frame_continue() {
+        let (schedule, _world) = setup();
+
+        let mut stepping = Stepping::new();
+        stepping
+            .add_schedule(TestSchedule)
+            .enable()
+            .always_run(TestSchedule, second_system)
+            .continue_frame();
+
+        // start a new frame, then run the schedule three times; first system
+        // should only run on the first one
+        stepping.next_frame();
+        assert_systems_run!(
+            &schedule,
+            stepping.skipped_systems(&schedule),
+            first_system,
+            second_system
+        );
+        assert_systems_run!(
+            &schedule,
+            stepping.skipped_systems(&schedule),
+            second_system
+        );
+    }
+    #[test]
+    fn multiple_calls_per_frame_step() {
+        let (schedule, _world) = setup();
+
+        let mut stepping = Stepping::new();
+        stepping.add_schedule(TestSchedule).enable().step_frame();
+
+        // start a new frame, then run the schedule three times; first system
```

Comment:
Three?

Reply:
Lol, lesson learned.  During first pass of writing tests, never commit to a number in the comments.  I'm cleaning these all up.